File: rifa1122/lib/features/rifas/models/categoria_rifa.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'categoria_rifa.freezed.dart';
part 'categoria_rifa.g.dart';

@freezed
class CategoriaRifa with _$CategoriaRifa {
  const factory CategoriaRifa({
    required String id,
    required String nombre,
    required String color,
    required int valorBoleta,
    required int totalRecaudo,
    required double rake,
    required int fondoPremios,
    required int premioPorGanador,
    String? comentario,
  }) = _CategoriaRifa;

  factory CategoriaRifa.fromJson(Map<String, dynamic> json) => _$CategoriaRifaFromJson(json);
}

/// Example JSON:
/// {
///   "id": "550e8400-e29b-41d4-a716-446655440002",
///   "nombre": "Bronce",
///   "color": "Marrón",
///   "valor_boleta": 5000,
///   "total_recaudo": 500000,
///   "rake": 0.25,
///   "fondo_premios": 375000,
///   "premio_por_ganador": 187500,
///   "comentario": "Premium"
/// }

File: backend/app/schemas/categoria_rifa.py
from pydantic import BaseModel
from typing import Optional
from decimal import Decimal
from uuid import UUID


class CategoriaRifaBase(BaseModel):
    nombre: str
    color: Optional[str] = None
    valor_boleta: int
    total_recaudo: Optional[int] = None
    rake: Optional[Decimal] = None
    fondo_premios: Optional[int] = None
    premio_por_ganador: Optional[int] = None
    comentario: Optional[str] = None


class CategoriaRifaCreate(CategoriaRifaBase):
    pass


class CategoriaRifaUpdate(BaseModel):
    nombre: Optional[str] = None
    color: Optional[str] = None
    valor_boleta: Optional[int] = None
    total_recaudo: Optional[int] = None
    rake: Optional[Decimal] = None
    fondo_premios: Optional[int] = None
    premio_por_ganador: Optional[int] = None
    comentario: Optional[str] = None


class CategoriaRifaOut(CategoriaRifaBase):
    id: UUID

    class Config:
        from_attributes = True